cpp/include/cudf/column/column.hpp

```diff
@@ -75,6 +76,21 @@ class column {
    */
   column(column&& other) noexcept;
 
+  /**
+   * @brief Move the contents from `other` to create a new column.
+   *
+   * After the move, `other.data() == NULL`
+   *
+   * @param other The device_uvector whose contents will be moved into the new column.
+   */
+  template <typename T>
+  column(rmm::device_uvector<T>&& other) noexcept
+    : _type{cudf::type_to_id<T>()},
+      _size{static_cast<cudf::size_type>(other.size())},
+      _data{other.release()}
```

Comment:
Let's use `buff` or `buffer` instead of `other`.

Reply:
In addition, since the buffer may have `size` exceeds `numeric_limits<size_type>`, we may need to check that:
```
CUDF_EXPECT(...., "Input buffer has size exceeds column capacity");
```

Reply:
As a note: adding that size check means this can't be `noexcept`.

Reply:
`device_uvector<T>` is not a `buffer`.

`other` is a standard, canonical name used for ctors like this.